kubernetes/crd/domain-v1beta1-crd.yaml

```diff
@@ -9160,6 +9307,54 @@ spec:
                               - path
                           required:
                           - name
+                      commonMount:
+                        description: The common mount with the containers hosting
+                          files to be copied to the common volume.
+                        type: object
+                        properties:
+                          sizeLimit:
+                            description: The emptyDir volume size limit. Defaults
+                              to unset.
+                            type: string
+                          mountPath:
+                            description: The common mount path. Defaults to /common.
+                            type: string
+                          targetPath:
+                            description: The target mount path. Defaults to '/tmpCommonMount'.
+                            type: string
+                          containers:
+                            description: The common mount containers.
+                            type: array
+                            items:
+                              type: object
+                              properties:
+                                image:
+                                  description: The common mount container image name.
+                                  type: string
+                                imagePullPolicy:
+                                  description: The image pull policy for the common
+                                    mount container image. Legal values are Always,
+                                    Never, and IfNotPresent. Defaults to Always if
+                                    image ends in :latest; IfNotPresent, otherwise.
+                                  type: string
+                                  enum:
+                                  - Always
+                                  - Never
+                                  - IfNotPresent
+                                name:
+                                  description: The name for the operator created init
+                                    container. Set to 'operator-common-container-${CONTAINER_NUM}'.
+                                  type: string
+                                command:
```

Comment:
An init container can have command and args.   Do we allow user to change the  command (which shell or others)? Also, we need to document how this is finally in the container spec.  For example,  user if coding the yaml manually can have:

command: ["/bin/sh"]
args: ["-c", "cp something somewhere"]

How this command translate into the above?

Reply:
The command that the user provides using this schema element translates to "cp something somewhere" (in args). I was envisioning use-cases where a user wants to perform additional tasks such as unzipping a file, copying files in a different location, or executing a script. We populate the `COMMON_MOUNT_PATH` and `COMMON_TARGET_PATH` env variables in the init container for this. I wanted to avoid having multiple moving parts where users can pass additional arguments to commands or change the command to something other than executing a shell command or script. I will look into the documentation part to clarify how the specified command finally translates in the container spec.

Reply:
I am trying to avoid confusion, if user tries to put the command as "/bin/bash -c 'blah blah'" .  If the user knows how it is translated then they can adjust accordingly.

Reply:
For a variety of reasons, I think we should have a wrapper script that in turn calls the specified commands.  If the user wants to see how a custom command is translated, they can examine the wrapper script. I just made a separate comment that details wrapper script ideas.